src/parser.rs

```diff
@@ -710,47 +726,26 @@ where
     e2: core::marker::PhantomData<E2>,
 }
 
-impl<F, G, I, O, O2, E, E2> TryMapCut<F, G, I, O, O2, E, E2>
-where
-    F: Parser<I, O, E>,
-    G: FnMut(O) -> Result<O2, E2>,
-    I: Stream,
-    E: FromExternalError<I, E2>,
-{
-    #[inline(always)]
-    fn new(parser: F, map: G) -> Self {
-        Self {
-            parser,
-            map,
-            i: Default::default(),
-            o: Default::default(),
-            o2: Default::default(),
-            e: Default::default(),
-            e2: Default::default(),
-        }
-    }
-}
-
-impl<F, G, I, O, O2, E, E2> Parser<I, O2, E> for TryMapCut<F, G, I, O, O2, E, E2>
+impl<F, G, I, O, O2, E, E2> Parser<I, O2, ErrMode<E>> for TryMapCut<F, G, I, O, O2, E, E2>
 where
-    F: Parser<I, O, E>,
+    F: ModalParser<I, O, E>,
     G: FnMut(O) -> Result<O2, E2>,
     I: Stream,
     E: FromExternalError<I, E2>,
 {
     #[inline]
-    fn parse_next(&mut self, input: &mut I) -> PResult<O2, E> {
+    fn parse_next(&mut self, input: &mut I) -> ModalResult<O2, E> {
         let start = input.checkpoint();
         let o = self.parser.parse_next(input)?;
 
         (self.map)(o).map_err(|err| {
             input.reset(&start);
-            ErrMode::from_external_error(input, ErrorKind::Verify, err).cut()
+            ErrMode::from_external_error(input, err).cut()
```

Comment:
@epage FYI how this crate upgrades to winnow 0.7 and retain the `try_map_cut` combinator.

Not sure if we can achieve the same behavior without `Cut`. From the changelog, it seems `cut` is secondary to use?

Reply:
Making `ErrMode` optional was about paying for what you use. `toml_edit` still uses `ModalResult` and `cut_err`.

To not use `cut_err`, you need to find ways to restructure a parser to reduce the use of `opt`/`alt` to where you don't need it (e.g. using `dispatch!` instead).

Reply:
Thanks for your information! Then I'd merge this patch as is.

I'd investigate if `dispatch!` can be used here later.

Reply:
https://github.com/winnow-rs/winnow/pull/692 is an example of that. I only did that so I wouldn't force `ErrMode` on users.

Reply:
Thanks for the reference.

After a closer look it seems cronexpr wants to terminate immediately if a part match is success but the value is an unexpected one, thus `try_map_cut`. No backtrace is expected here.

So I'd continue the current approach unless I find a way to express the semantic better.